### PR TITLE
feat(filter status): add all filter status

### PR DESCRIPTION
**Pull request by anitavincent**
#### What is the purpose of this pull request?
Add global loading, global error and empty filter status

#### How should this be manually tested?
Check status stories

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Quality improvement (tests or refactors)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (small fix or feature that doesn't impact functionalities)
- [ ] Requires change to documentation, which has been updated accordingly

### CODE REVIEW

File: packages/admin-ui/src/filters/filter.style.tsx [hunk 1]
```diff
@@ -59,7 +59,19 @@ export const disclosureStatusLabel = style({
...
+export const searchbox = ({
+  omitBottomPadding,
+}: {
+  omitBottomPadding: boolean
+}) =>
+  style({
+    margin: '$l',
+    marginBottom: omitBottomPadding ? '0' : undefined,
+    order: -1,
+    '& ~ .__admin-ui-filter-status': { minHeight: '16em' },
```
Comment (lucasaarcoverde) [inline packages/admin-ui/src/filters/filter.style.tsx:71]:
```suggestion
    '& ~ .__admin-ui-filter-status': { minHeight: '16rem' },
```

File: packages/admin-ui/src/filters/filter.style.tsx [hunk 2]
```diff
@@ -59,7 +59,19 @@ export const disclosureStatusLabel = style({
...
+  })
```
Comment (lucasaarcoverde) [inline packages/admin-ui/src/filters/filter.style.tsx:72]: The prop is `omitBottomPadding` but you are setting the `marginBottom` is it right?
```suggestion

export const searchbox = (omitBottomPadding: boolean) =>
  style({
    margin: '$l',
    marginBottom: omitBottomPadding ? '0' : undefined,
    order: -1,
    '& ~ .__admin-ui-filter-status': { minHeight: '16em' },
  })
```

File: packages/admin-ui/src/filters/filter.style.tsx [hunk 3]
```diff
@@ -59,7 +59,19 @@ export const disclosureStatusLabel = style({
...
+
+export const errorStatusLayout = style({ padding: '$l' })
```
Comment (lucasaarcoverde) [inline packages/admin-ui/src/filters/filter.style.tsx:74]:
```suggestion
export const statusContainer = style({ padding: '$l' })
```

File: packages/admin-ui/src/filters/filter-status.tsx [hunk 4]
```diff
@@ -22,8 +24,31 @@ export function FilterStatus(props: FilterStatusProps) {
 
   return (
     <>
+      {status === 'empty' && (
+        <Box csx={{ padding: '$l' }}>
```
Comment (lucasaarcoverde) [inline packages/admin-ui/src/filters/filter-status.tsx:28]:
```suggestion
        <Box csx={styles.statusContainer}>
```

File: packages/admin-ui/src/filters/filter-status.tsx [hunk 5]
```diff
@@ -22,8 +24,31 @@ export function FilterStatus(props: FilterStatusProps) {
...
+          <Text variant="title2">{formatMessage('empty')}</Text>
+        </Box>
+      )}
+
+      {status === 'loading' && (
+        <Box csx={{ padding: '$l' }}>
```
Comment (lucasaarcoverde) [inline packages/admin-ui/src/filters/filter-status.tsx:34]:
```suggestion
        <Stack csx={styles.statusContainer} space="$2xl">
```

File: packages/admin-ui/src/filters/filter-status.tsx [hunk 6]
```diff
@@ -22,8 +24,31 @@ export function FilterStatus(props: FilterStatusProps) {
...
+          <Stack space="$xl" csx={{ width: '3/4' }}>
```
Comment (lucasaarcoverde) [inline packages/admin-ui/src/filters/filter-status.tsx:35]:
```suggestion
          <Stack space="$xl" csx={{ width: '3/4', '> *': { height: 20 } }}>
```

File: packages/admin-ui/src/filters/filter-status.tsx [hunk 7]
```diff
@@ -22,8 +24,31 @@ export function FilterStatus(props: FilterStatusProps) {
...
+            <Skeleton csx={{ height: 20, width: '9/12' }} />
+            <Skeleton csx={{ height: 20, width: '10/12' }} />
+            <Skeleton csx={{ height: 20, width: '11/12' }} />
+            <Skeleton csx={{ height: 20, width: '8/12' }} />
+          </Stack>
+
+          <Flex justify="end" csx={{ marginTop: '$2xl' }}>
+            <Skeleton csx={{ height: 32, width: 62, marginRight: '$m' }} />
+            <Skeleton csx={{ height: 32, width: 62 }} />
+          </Flex>
```
Comment (lucasaarcoverde) [inline packages/admin-ui/src/filters/filter-status.tsx:46]:
```suggestion
          <Stack direction="row" csx={{ '> *': { height: 32, width: 62 }, justifyContent: 'flex-end' }}>
            <Skeleton />
            <Skeleton />
          </Stack>
```

File: packages/admin-ui/src/filters/filter-status.tsx [hunk 8]
```diff
@@ -32,7 +57,7 @@ export function FilterStatus(props: FilterStatusProps) {
       )}
 
       {status === 'not-found' && (
-        <Box csx={style.searchStatusLayout}>
+        <Box className="__admin-ui-filter-status">
           <Stack space="$s">
```
Comment (lucasaarcoverde) [inline packages/admin-ui/src/filters/filter-status.tsx:61]:
```suggestion
        <Stack space="$s" className="__admin-ui-filter-status">
```

File: packages/admin-ui/src/filters/filter.style.tsx (hunk 2 above)
Comment (anitavincent) [reply packages/admin-ui/src/filters/filter.style.tsx:72]: true

File: packages/admin-ui/src/filters/filter-status.tsx [hunk 9]
```diff
@@ -25,30 +24,38 @@ export function FilterStatus(props: FilterStatusProps) {
...
           </Stack>
 
-          <Flex justify="end" csx={{ marginTop: '$2xl' }}>
-            <Skeleton csx={{ height: 32, width: 62, marginRight: '$m' }} />
-            <Skeleton csx={{ height: 32, width: 62 }} />
-          </Flex>
-        </Box>
+          <Stack
+            direction="row"
+            space="$m"
```
Comment (lucasaarcoverde) [inline packages/admin-ui/src/filters/filter-status.tsx:44]: is already the default value

File: packages/admin-ui/src/filters/filter.style.tsx [hunk 10]
```diff
@@ -60,18 +60,18 @@ export const disclosureStatusLabel = style({
 })
 
 export const searchbox = ({
-  omitBottomPadding,
+  omitBottomMargin,
 }: {
-  omitBottomPadding: boolean
+  omitBottomMargin: boolean
 }) =>
```
Comment (lucasaarcoverde) [inline packages/admin-ui/src/filters/filter.style.tsx:66]: IMO is not necessary to be an object, you can pass the single argument directly

**Review by lucasaarcoverde — APPROVED** [review]
LGTM

File: packages/admin-ui/src/filters/__stories__/filter-status.stories.tsx [hunk 11]
```diff
@@ -122,9 +128,15 @@ export function ErrorSearchExample() {
...
+  const stateSimple = useFilterMultipleState()
+
+  useEffect(() => {
+    stateSimple.setStatus('error')
+  }, [])
+
   return (
-    <>
-      <FilterDisclosure state={state}>Controlled empty state</FilterDisclosure>
+    <Stack space="$m">
```
Comment (matheusps) [inline packages/admin-ui/src/filters/__stories__/filter-status.stories.tsx:138]:
```suggestion
    <Stack>
```

File: packages/admin-ui/src/filters/filter-status.tsx [hunk 12]
```diff
@@ -22,8 +23,38 @@ export function FilterStatus(props: FilterStatusProps) {
...
+          <Stack space="$xl" csx={{ width: '3/4' }}>
+            <Skeleton csx={{ height: 20 }} />
+            <Skeleton csx={{ width: '9/12', height: 20 }} />
+            <Skeleton csx={{ width: '10/12', height: 20 }} />
+            <Skeleton csx={{ width: '11/12', height: 20 }} />
+            <Skeleton csx={{ width: '8/12', height: 20 }} />
+          </Stack>
+
+          <Stack
+            direction="row"
```
Comment (matheusps) [inline packages/admin-ui/src/filters/filter-status.tsx:43]: Shouldn't it be an `Inline` then?

File: packages/admin-ui/src/filters/filter.style.tsx [hunk 13]
```diff
@@ -59,7 +59,19 @@ export const disclosureStatusLabel = style({
   maxWidth: '300px',
 })
 
-export const searchStatusLayout = style({ marginX: '$xl', height: '16em' })
+export const searchbox = ({
```
Comment (matheusps) [inline packages/admin-ui/src/filters/filter.style.tsx:62]: This is a style variant

File: packages/admin-ui/src/filters/filter-status.tsx (hunk 12 above)
Comment (anitavincent) [reply packages/admin-ui/src/filters/filter-status.tsx:43]: Inline didn't work with the child selector

**Review by matheusps — APPROVED** [review]

**Pull request merged**